log4j-core/src/main/java/org/apache/logging/log4j/core/config/builder/api/ConfigurationBuilder.java

```diff
@@ -61,6 +61,10 @@ public interface ConfigurationBuilder<T extends Configuration> extends Builder<T
      */
     ConfigurationBuilder<T> add(CustomLevelComponentBuilder builder);
 
+    default ConfigurationBuilder<T> add(MonitorResourceComponentBuilder builder) {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
1. Add a generic method for top-level elements in `ConfigurationBuilder`:
    ```java
    ConfigurationBuilder<T> addComponent(ComponentBuilder<?> builder);
    ```
    It mirrors `ComponentBuilder#addComponent`, but uses a distinct name to differentiate from specialized `add` methods.